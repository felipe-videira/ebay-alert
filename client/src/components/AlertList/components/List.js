import './List.scss';
import React from 'react';
import { Empty, Button, Layout, List, Skeleton } from 'antd';

export default ({ 
    data = [], 
    loading = false, 
    onCreate = () => {}, 
    onEdit = (item) => {}, 
    onDelete = (item) => {}, 
}) => {
    return (
      <Layout className="alert-list">
        {!data.length 
          ? <Empty 
              className="alert-list__empty"
              image={Empty.PRESENTED_IMAGE_SIMPLE}
              description="You have no alerts"
            >
              <Button 
                type="primary"
                size="large"
                onClick={() => onCreate()} 
              >
                Create one
              </Button>
            </Empty>
          : <List
              className="alert-list"
              loading={loading}
              itemLayout="horizontal"
              dataSource={data}
              renderItem={item => (
                <List.Item
                  actions={[
                    <Button 
                      shape="circle" 
                      icon="edit"
                      onClick={() => onEdit(item)} 
                    />,
                    <Button 
                      shape="circle" 
                      icon="delete" 
                      onClick={() => onDelete(item)} 
                    /> 
                  ]}
                >
                  <Skeleton 
                    title={false} 
                    loading={item.loading} 
                    active
                  >
                    <List.Item.Meta
                      title={`"${item.searchPhrase}"`}
                      description={`send to ${item.email}`}
                    />
                    <div>{`every ${item.frequency} minutes`}</div>
                  </Skeleton>
                </List.Item>
              )}
          />}
        {!!data.length && 
        <Button 
          type="primary" 
          shape="circle" 
          icon="plus"
          className="alert-list__create-button" 
          onClick={() => onCreate()} 
        ></Button>}
      </Layout>
    );
}

 
