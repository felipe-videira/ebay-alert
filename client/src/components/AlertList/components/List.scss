@import 'assets/styles/mixins';
@import 'assets/styles/variables';
@import 'assets/styles/base';

.alert-list {
    min-height: 100%;
    background-color: $white;
    padding: 5%;

    &__list {
        padding: 5%;
    }

    &__create-btn {
        @extend %responsive-btn;

        @include breakpoint(not-mobile) {
            width: 10%;
            align-self: flex-end;
        }

        @include breakpoint(mobile) {
            font-size: 26px;
            position: fixed;
            right: 5%;
            bottom: 10%;
        }
    }

    &__action {
        @include breakpoint(mobile) {
            display: none;
        }
    }

    &__load-more-btn {
        text-align: center;
        margin: 5% 40%;
        width: 20%;
    }

    &__empty-state {
        color: $disabled-color;
        font-size: 16px;
        margin: 10% 0;
    }
}