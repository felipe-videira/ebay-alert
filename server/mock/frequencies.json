[
    {
        "value": "2",
        "label": "2"
    },  
    {
        "value": "10",
        "label": "10"
    },  
    {
        "value": "30",
        "label": "30"
    }  
]
