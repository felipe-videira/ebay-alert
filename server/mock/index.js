const Frequency = require('../src/models/frequency');
const Locale = require('../src/models/locale');
const frequencies = require('./frequencies.json');
const locales = require('./locales.json');

Frequency.exists().then(exists => {
    !exists && Frequency.insertMany(frequencies);
});

Locale.exists().then(exists => {
    !exists && Locale.insertMany(locales);
});


