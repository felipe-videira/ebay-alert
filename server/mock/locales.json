[
    {
        "lng": "pt",
        "translation": {
            "emailLabel": "E-mail"
        }
    },
    {
        "lng": "en",
        "translation": {
            "emailLabel": "E-mail"
        }
    }
]