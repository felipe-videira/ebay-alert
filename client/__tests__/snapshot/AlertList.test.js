import React from 'react';
import renderer from 'react-test-renderer';
import mock from '../../__mocks__/data/alertList';
import List from '../../src/components/AlertList/components/List';

describe('AlertList snapshot tests', () => {
  it('renders correctly on default values', () => {
    expect(renderer.create(
      <List />
    ).toJSON())
    .toMatchSnapshot();
  });
  it('renders correctly with data', () => {
    expect(renderer.create(
      <List data={mock.data} />
    ).toJSON())
    .toMatchSnapshot();
  });
  it('renders correctly with data and custom keys', () => {
    expect(renderer.create(
      <List 
        data={mock.dataWithCustomKeys}
        titleKey={mock.titleKey}
        descriptionKey={mock.descriptionKey}
        infoKey={mock.infoKey}
      />
    ).toJSON())
    .toMatchSnapshot();
  });
  it('renders correctly on loading', () => {
    expect(renderer.create(
      <List loading />
    ).toJSON())
    .toMatchSnapshot();
  });
  it('renders correctly on mobile mode', () => {
    expect(renderer.create(
      <List mobile />
    ).toJSON())
    .toMatchSnapshot();
  });
  it('renders correctly when has more to load', () => {
    expect(renderer.create(
      <List hasMoreToLoad />
    ).toJSON())
    .toMatchSnapshot();
  });
});
