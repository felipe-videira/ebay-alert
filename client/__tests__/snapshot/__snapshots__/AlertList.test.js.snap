// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AlertList snapshot tests renders correctly on default values 1`] = `
<section
  className="alert-list ant-layout"
>
  <div
    className="ant-empty ant-empty-normal alert-list__empty"
  >
    <div
      className="ant-empty-image"
    >
      <svg
        height="41"
        viewBox="0 0 64 41"
        width="64"
        xmlns="http://www.w3.org/2000/svg"
      >
        <g
          fill="none"
          fillRule="evenodd"
          transform="translate(0 1)"
        >
          <ellipse
            cx="32"
            cy="33"
            fill="#F5F5F5"
            rx="32"
            ry="7"
          />
          <g
            fillRule="nonzero"
            stroke="#D9D9D9"
          >
            <path
              d="M55 12.76L44.854 1.258C44.367.474 43.656 0 42.907 0H21.093c-.749 0-1.46.474-1.947 1.257L9 12.761V22h46v-9.24z"
            />
            <path
              d="M41.613 15.931c0-1.605.994-2.93 2.227-2.931H55v18.137C55 33.26 53.68 35 52.05 35h-40.1C10.32 35 9 33.259 9 31.137V13h11.16c1.233 0 2.227 1.323 2.227 2.928v.022c0 1.605 1.005 2.901 2.237 2.901h14.752c1.232 0 2.237-1.308 2.237-2.913v-.007z"
              fill="#FAFAFA"
            />
          </g>
        </g>
      </svg>
    </div>
    <p
      className="ant-empty-description"
    >
      alertListEmpty
    </p>
    <div
      className="ant-empty-footer"
    >
      <button
        className="ant-btn ant-btn-primary ant-btn-lg"
        onClick={[Function]}
        type="button"
      >
        <span>
          createButton
        </span>
      </button>
    </div>
  </div>
</section>
`;

exports[`AlertList snapshot tests renders correctly on loading 1`] = `
<section
  className="alert-list ant-layout"
>
  <div
    className="ant-empty ant-empty-normal alert-list__empty"
  >
    <div
      className="ant-empty-image"
    >
      <svg
        height="41"
        viewBox="0 0 64 41"
        width="64"
        xmlns="http://www.w3.org/2000/svg"
      >
        <g
          fill="none"
          fillRule="evenodd"
          transform="translate(0 1)"
        >
          <ellipse
            cx="32"
            cy="33"
            fill="#F5F5F5"
            rx="32"
            ry="7"
          />
          <g
            fillRule="nonzero"
            stroke="#D9D9D9"
          >
            <path
              d="M55 12.76L44.854 1.258C44.367.474 43.656 0 42.907 0H21.093c-.749 0-1.46.474-1.947 1.257L9 12.761V22h46v-9.24z"
            />
            <path
              d="M41.613 15.931c0-1.605.994-2.93 2.227-2.931H55v18.137C55 33.26 53.68 35 52.05 35h-40.1C10.32 35 9 33.259 9 31.137V13h11.16c1.233 0 2.227 1.323 2.227 2.928v.022c0 1.605 1.005 2.901 2.237 2.901h14.752c1.232 0 2.237-1.308 2.237-2.913v-.007z"
              fill="#FAFAFA"
            />
          </g>
        </g>
      </svg>
    </div>
    <p
      className="ant-empty-description"
    >
      alertListEmpty
    </p>
    <div
      className="ant-empty-footer"
    >
      <button
        className="ant-btn ant-btn-primary ant-btn-lg"
        onClick={[Function]}
        type="button"
      >
        <span>
          createButton
        </span>
      </button>
    </div>
  </div>
</section>
`;

exports[`AlertList snapshot tests renders correctly on mobile mode 1`] = `
<section
  className="alert-list ant-layout"
>
  <div
    className="ant-empty ant-empty-normal alert-list__empty"
  >
    <div
      className="ant-empty-image"
    >
      <svg
        height="41"
        viewBox="0 0 64 41"
        width="64"
        xmlns="http://www.w3.org/2000/svg"
      >
        <g
          fill="none"
          fillRule="evenodd"
          transform="translate(0 1)"
        >
          <ellipse
            cx="32"
            cy="33"
            fill="#F5F5F5"
            rx="32"
            ry="7"
          />
          <g
            fillRule="nonzero"
            stroke="#D9D9D9"
          >
            <path
              d="M55 12.76L44.854 1.258C44.367.474 43.656 0 42.907 0H21.093c-.749 0-1.46.474-1.947 1.257L9 12.761V22h46v-9.24z"
            />
            <path
              d="M41.613 15.931c0-1.605.994-2.93 2.227-2.931H55v18.137C55 33.26 53.68 35 52.05 35h-40.1C10.32 35 9 33.259 9 31.137V13h11.16c1.233 0 2.227 1.323 2.227 2.928v.022c0 1.605 1.005 2.901 2.237 2.901h14.752c1.232 0 2.237-1.308 2.237-2.913v-.007z"
              fill="#FAFAFA"
            />
          </g>
        </g>
      </svg>
    </div>
    <p
      className="ant-empty-description"
    >
      alertListEmpty
    </p>
    <div
      className="ant-empty-footer"
    >
      <button
        className="ant-btn ant-btn-primary ant-btn-lg"
        onClick={[Function]}
        type="button"
      >
        <span>
          createButton
        </span>
      </button>
    </div>
  </div>
</section>
`;

exports[`AlertList snapshot tests renders correctly when has more to load 1`] = `
<section
  className="alert-list ant-layout"
>
  <div
    className="ant-empty ant-empty-normal alert-list__empty"
  >
    <div
      className="ant-empty-image"
    >
      <svg
        height="41"
        viewBox="0 0 64 41"
        width="64"
        xmlns="http://www.w3.org/2000/svg"
      >
        <g
          fill="none"
          fillRule="evenodd"
          transform="translate(0 1)"
        >
          <ellipse
            cx="32"
            cy="33"
            fill="#F5F5F5"
            rx="32"
            ry="7"
          />
          <g
            fillRule="nonzero"
            stroke="#D9D9D9"
          >
            <path
              d="M55 12.76L44.854 1.258C44.367.474 43.656 0 42.907 0H21.093c-.749 0-1.46.474-1.947 1.257L9 12.761V22h46v-9.24z"
            />
            <path
              d="M41.613 15.931c0-1.605.994-2.93 2.227-2.931H55v18.137C55 33.26 53.68 35 52.05 35h-40.1C10.32 35 9 33.259 9 31.137V13h11.16c1.233 0 2.227 1.323 2.227 2.928v.022c0 1.605 1.005 2.901 2.237 2.901h14.752c1.232 0 2.237-1.308 2.237-2.913v-.007z"
              fill="#FAFAFA"
            />
          </g>
        </g>
      </svg>
    </div>
    <p
      className="ant-empty-description"
    >
      alertListEmpty
    </p>
    <div
      className="ant-empty-footer"
    >
      <button
        className="ant-btn ant-btn-primary ant-btn-lg"
        onClick={[Function]}
        type="button"
      >
        <span>
          createButton
        </span>
      </button>
    </div>
  </div>
</section>
`;

exports[`AlertList snapshot tests renders correctly with data 1`] = `
<section
  className="alert-list ant-layout"
>
  <button
    className="ant-btn alert-list__create-btn"
    onClick={[Function]}
    type="button"
  >
    <i
      aria-label="icon: plus"
      className="anticon anticon-plus"
    >
      <svg
        aria-hidden="true"
        className=""
        data-icon="plus"
        fill="currentColor"
        focusable="false"
        height="1em"
        viewBox="64 64 896 896"
        width="1em"
      >
        <path
          d="M482 152h60q8 0 8 8v704q0 8-8 8h-60q-8 0-8-8V160q0-8 8-8z"
        />
        <path
          d="M176 474h672q8 0 8 8v60q0 8-8 8H176q-8 0-8-8v-60q0-8 8-8z"
        />
      </svg>
    </i>
    <p>
      createButton
       
    </p>
  </button>
  <div
    className="ant-list alert-list__list ant-list-split"
  >
    <div
      className="ant-spin-nested-loading"
    >
      <div
        className="ant-spin-container"
      >
        <ul
          className="ant-list-items"
        >
          <li
            className="ant-list-item"
            onClick={[Function]}
          >
            <div
              className="ant-list-item-meta"
            >
              <div
                className="ant-list-item-meta-content"
              >
                <h4
                  className="ant-list-item-meta-title"
                >
                  alertListTitle
                </h4>
                <div
                  className="ant-list-item-meta-description"
                >
                  alertListDescription
                </div>
              </div>
            </div>
            <div>
              alertListInfo
            </div>
            <ul
              className="ant-list-item-action"
            >
              <li>
                <button
                  className="ant-btn alert-list__action ant-btn-circle ant-btn-icon-only"
                  onClick={[Function]}
                  type="button"
                >
                  <i
                    aria-label="icon: edit"
                    className="anticon anticon-edit"
                  >
                    <svg
                      aria-hidden="true"
                      className=""
                      data-icon="edit"
                      fill="currentColor"
                      focusable="false"
                      height="1em"
                      viewBox="64 64 896 896"
                      width="1em"
                    >
                      <path
                        d="M257.7 752c2 0 4-.2 6-.5L431.9 722c2-.4 3.9-1.3 5.3-2.8l423.9-423.9a9.96 9.96 0 0 0 0-14.1L694.9 114.9c-1.9-1.9-4.4-2.9-7.1-2.9s-5.2 1-7.1 2.9L256.8 538.8c-1.5 1.5-2.4 3.3-2.8 5.3l-29.5 168.2a33.5 33.5 0 0 0 9.4 29.8c6.6 6.4 14.9 9.9 23.8 9.9zm67.4-174.4L687.8 215l73.3 73.3-362.7 362.6-88.9 15.7 15.6-89zM880 836H144c-17.7 0-32 14.3-32 32v36c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-36c0-17.7-14.3-32-32-32z"
                      />
                    </svg>
                  </i>
                </button>
                <em
                  className="ant-list-item-action-split"
                />
              </li>
              <li>
                <button
                  className="ant-btn alert-list__action ant-btn-circle ant-btn-icon-only"
                  onClick={[Function]}
                  type="button"
                >
                  <i
                    aria-label="icon: delete"
                    className="anticon anticon-delete"
                  >
                    <svg
                      aria-hidden="true"
                      className=""
                      data-icon="delete"
                      fill="currentColor"
                      focusable="false"
                      height="1em"
                      viewBox="64 64 896 896"
                      width="1em"
                    >
                      <path
                        d="M360 184h-8c4.4 0 8-3.6 8-8v8h304v-8c0 4.4 3.6 8 8 8h-8v72h72v-80c0-35.3-28.7-64-64-64H352c-35.3 0-64 28.7-64 64v80h72v-72zm504 72H160c-17.7 0-32 14.3-32 32v32c0 4.4 3.6 8 8 8h60.4l24.7 523c1.6 34.1 29.8 61 63.9 61h454c34.2 0 62.3-26.8 63.9-61l24.7-523H888c4.4 0 8-3.6 8-8v-32c0-17.7-14.3-32-32-32zM731.3 840H292.7l-24.2-512h487l-24.2 512z"
                      />
                    </svg>
                  </i>
                </button>
              </li>
            </ul>
          </li>
          <li
            className="ant-list-item"
            onClick={[Function]}
          >
            <div
              className="ant-list-item-meta"
            >
              <div
                className="ant-list-item-meta-content"
              >
                <h4
                  className="ant-list-item-meta-title"
                >
                  alertListTitle
                </h4>
                <div
                  className="ant-list-item-meta-description"
                >
                  alertListDescription
                </div>
              </div>
            </div>
            <div>
              alertListInfo
            </div>
            <ul
              className="ant-list-item-action"
            >
              <li>
                <button
                  className="ant-btn alert-list__action ant-btn-circle ant-btn-icon-only"
                  onClick={[Function]}
                  type="button"
                >
                  <i
                    aria-label="icon: edit"
                    className="anticon anticon-edit"
                  >
                    <svg
                      aria-hidden="true"
                      className=""
                      data-icon="edit"
                      fill="currentColor"
                      focusable="false"
                      height="1em"
                      viewBox="64 64 896 896"
                      width="1em"
                    >
                      <path
                        d="M257.7 752c2 0 4-.2 6-.5L431.9 722c2-.4 3.9-1.3 5.3-2.8l423.9-423.9a9.96 9.96 0 0 0 0-14.1L694.9 114.9c-1.9-1.9-4.4-2.9-7.1-2.9s-5.2 1-7.1 2.9L256.8 538.8c-1.5 1.5-2.4 3.3-2.8 5.3l-29.5 168.2a33.5 33.5 0 0 0 9.4 29.8c6.6 6.4 14.9 9.9 23.8 9.9zm67.4-174.4L687.8 215l73.3 73.3-362.7 362.6-88.9 15.7 15.6-89zM880 836H144c-17.7 0-32 14.3-32 32v36c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-36c0-17.7-14.3-32-32-32z"
                      />
                    </svg>
                  </i>
                </button>
                <em
                  className="ant-list-item-action-split"
                />
              </li>
              <li>
                <button
                  className="ant-btn alert-list__action ant-btn-circle ant-btn-icon-only"
                  onClick={[Function]}
                  type="button"
                >
                  <i
                    aria-label="icon: delete"
                    className="anticon anticon-delete"
                  >
                    <svg
                      aria-hidden="true"
                      className=""
                      data-icon="delete"
                      fill="currentColor"
                      focusable="false"
                      height="1em"
                      viewBox="64 64 896 896"
                      width="1em"
                    >
                      <path
                        d="M360 184h-8c4.4 0 8-3.6 8-8v8h304v-8c0 4.4 3.6 8 8 8h-8v72h72v-80c0-35.3-28.7-64-64-64H352c-35.3 0-64 28.7-64 64v80h72v-72zm504 72H160c-17.7 0-32 14.3-32 32v32c0 4.4 3.6 8 8 8h60.4l24.7 523c1.6 34.1 29.8 61 63.9 61h454c34.2 0 62.3-26.8 63.9-61l24.7-523H888c4.4 0 8-3.6 8-8v-32c0-17.7-14.3-32-32-32zM731.3 840H292.7l-24.2-512h487l-24.2 512z"
                      />
                    </svg>
                  </i>
                </button>
              </li>
            </ul>
          </li>
          <li
            className="ant-list-item"
            onClick={[Function]}
          >
            <div
              className="ant-list-item-meta"
            >
              <div
                className="ant-list-item-meta-content"
              >
                <h4
                  className="ant-list-item-meta-title"
                >
                  alertListTitle
                </h4>
                <div
                  className="ant-list-item-meta-description"
                >
                  alertListDescription
                </div>
              </div>
            </div>
            <div>
              alertListInfo
            </div>
            <ul
              className="ant-list-item-action"
            >
              <li>
                <button
                  className="ant-btn alert-list__action ant-btn-circle ant-btn-icon-only"
                  onClick={[Function]}
                  type="button"
                >
                  <i
                    aria-label="icon: edit"
                    className="anticon anticon-edit"
                  >
                    <svg
                      aria-hidden="true"
                      className=""
                      data-icon="edit"
                      fill="currentColor"
                      focusable="false"
                      height="1em"
                      viewBox="64 64 896 896"
                      width="1em"
                    >
                      <path
                        d="M257.7 752c2 0 4-.2 6-.5L431.9 722c2-.4 3.9-1.3 5.3-2.8l423.9-423.9a9.96 9.96 0 0 0 0-14.1L694.9 114.9c-1.9-1.9-4.4-2.9-7.1-2.9s-5.2 1-7.1 2.9L256.8 538.8c-1.5 1.5-2.4 3.3-2.8 5.3l-29.5 168.2a33.5 33.5 0 0 0 9.4 29.8c6.6 6.4 14.9 9.9 23.8 9.9zm67.4-174.4L687.8 215l73.3 73.3-362.7 362.6-88.9 15.7 15.6-89zM880 836H144c-17.7 0-32 14.3-32 32v36c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-36c0-17.7-14.3-32-32-32z"
                      />
                    </svg>
                  </i>
                </button>
                <em
                  className="ant-list-item-action-split"
                />
              </li>
              <li>
                <button
                  className="ant-btn alert-list__action ant-btn-circle ant-btn-icon-only"
                  onClick={[Function]}
                  type="button"
                >
                  <i
                    aria-label="icon: delete"
                    className="anticon anticon-delete"
                  >
                    <svg
                      aria-hidden="true"
                      className=""
                      data-icon="delete"
                      fill="currentColor"
                      focusable="false"
                      height="1em"
                      viewBox="64 64 896 896"
                      width="1em"
                    >
                      <path
                        d="M360 184h-8c4.4 0 8-3.6 8-8v8h304v-8c0 4.4 3.6 8 8 8h-8v72h72v-80c0-35.3-28.7-64-64-64H352c-35.3 0-64 28.7-64 64v80h72v-72zm504 72H160c-17.7 0-32 14.3-32 32v32c0 4.4 3.6 8 8 8h60.4l24.7 523c1.6 34.1 29.8 61 63.9 61h454c34.2 0 62.3-26.8 63.9-61l24.7-523H888c4.4 0 8-3.6 8-8v-32c0-17.7-14.3-32-32-32zM731.3 840H292.7l-24.2-512h487l-24.2 512z"
                      />
                    </svg>
                  </i>
                </button>
              </li>
            </ul>
          </li>
        </ul>
      </div>
    </div>
  </div>
</section>
`;

exports[`AlertList snapshot tests renders correctly with data and custom keys 1`] = `
<section
  className="alert-list ant-layout"
>
  <div
    className="ant-empty ant-empty-normal alert-list__empty"
  >
    <div
      className="ant-empty-image"
    >
      <svg
        height="41"
        viewBox="0 0 64 41"
        width="64"
        xmlns="http://www.w3.org/2000/svg"
      >
        <g
          fill="none"
          fillRule="evenodd"
          transform="translate(0 1)"
        >
          <ellipse
            cx="32"
            cy="33"
            fill="#F5F5F5"
            rx="32"
            ry="7"
          />
          <g
            fillRule="nonzero"
            stroke="#D9D9D9"
          >
            <path
              d="M55 12.76L44.854 1.258C44.367.474 43.656 0 42.907 0H21.093c-.749 0-1.46.474-1.947 1.257L9 12.761V22h46v-9.24z"
            />
            <path
              d="M41.613 15.931c0-1.605.994-2.93 2.227-2.931H55v18.137C55 33.26 53.68 35 52.05 35h-40.1C10.32 35 9 33.259 9 31.137V13h11.16c1.233 0 2.227 1.323 2.227 2.928v.022c0 1.605 1.005 2.901 2.237 2.901h14.752c1.232 0 2.237-1.308 2.237-2.913v-.007z"
              fill="#FAFAFA"
            />
          </g>
        </g>
      </svg>
    </div>
    <p
      className="ant-empty-description"
    >
      alertListEmpty
    </p>
    <div
      className="ant-empty-footer"
    >
      <button
        className="ant-btn ant-btn-primary ant-btn-lg"
        onClick={[Function]}
        type="button"
      >
        <span>
          createButton
        </span>
      </button>
    </div>
  </div>
</section>
`;
